extensions/barrage/src/main/java/io/deephaven/extensions/barrage/util/BarrageUtil.java

```diff
@@ -105,6 +112,73 @@ public class BarrageUtil {
     private static final String ATTR_TYPE_TAG = "type";
     private static final String ATTR_COMPONENT_TYPE_TAG = "componentType";
 
+    private static final boolean ENFORCE_FLATBUFFER_VERSION_CHECK =
+            Configuration.getInstance().getBooleanWithDefault("barrage.version.check", true);
+
+    static {
+        verifyFlatbufferCompatibility(Message.class);
+        verifyFlatbufferCompatibility(BarrageMessageWrapper.class);
+    }
+
+    private static void verifyFlatbufferCompatibility(Class<?> clazz) {
+        try {
+            clazz.getMethod("ValidateVersion").invoke(null);
+        } catch (InvocationTargetException e) {
+            Throwable targetException = e.getTargetException();
+            if (targetException instanceof NoSuchMethodError) {
+                // Caused when the reflective method is found and cannot be used because the flatbuffer version doesn't
+                // match
+                String requiredVersion = extractFlatBufferVersion(targetException.getMessage())
+                        .orElseThrow(() -> new UncheckedDeephavenException(
+                                "FlatBuffers version mismatch, can't read expected version", targetException));
+                Optional<String> foundVersion = Arrays.stream(Constants.class.getDeclaredMethods())
+                        .map(Method::getName)
+                        .map(BarrageUtil::extractFlatBufferVersion)
+                        .filter(Optional::isPresent)
+                        .map(Optional::get)
+                        .findFirst();
```

Comment:
At the risk of being even more over engineered, do we care about `java.lang.Package#getImplementationVersion`? (Fine if answer is "no")